Feature that allows players to milk goats for goats milk and make feta cheese.

grhmhome opened the pull request:
[Changelogs]:

:cl: Dr. Chef
add: Goat udders as a new udder subtype
add: Milking goats gives you goats milk
add: Goat milk can be turned into Feta Cheese. If you know how to make tofu and Cheddar, you can make Feta Cheese. 40 units of goat milk and 5 units of enzyme makes Feta Cheese. 
del: Old udder code that made milking goats for goat milk impossible
tweak: Tweaked old udder code so you can create udder subtypes more easily for different critters.
imageadd: Modifies food.dmi. Adds modified cheese icons for feta cheese
/:cl:

[why]: I wanted players to be able to obtain goats milk from goats. Previously, goats made cow milk, but now they make goat milk. This will give the chef new items to play around with and in the future could inspire new recipes that use goat milk and Feta Cheese.

![goatmilkfetacheesetest2001](https://user-images.githubusercontent.com/8340147/39460429-21ad68c0-4cb8-11e8-8876-58e61527d3b7.png)

grhmhome committed:
Adds the Feta cheese icons

grhmhome committed:
Update snacks_other.dm with Feta Cheese and slices

Adds Feta cheese wheel and slices.

grhmhome committed:
Updates drink_reagents.dm with goat milk

Goat milk exists now. Subtype of milk.

grhmhome committed:
Updates farm_animals.dm adds goat udders

This allows goats (and leaves the option for new critters) to be milked. Adds the milk type variable. Goats make goat milk now, instead of cows milk.

grhmhome committed:
Updates gutlunch.dm

I didn't know gutlunches had a type of udder, but they do. I had to change this code because I modified the udder code in farm_animals.dm. The overall changes allow you to make udder sub types to mobs and use different reagents.

grhmhome committed:
Update drink_reagents.dm milk is from cows

The description was changed so it shows that milk is from cows.

grhmhome committed:
Update food_mixtures.dm Feta Cheese recipe

Mixing 40 units of goat milk and 5 units of enzyme makes Feta Cheese. Creates Feta Cheese and slicing a wheel makes Feta Cheese slices.

ma44 commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm:
```diff
@@ -123,6 +123,7 @@
 //Gutlunch udder
 /obj/item/udder/gutlunch
 	name = "nutrient sac"
+	milktype = "milk"
```
The gutlunch makes two different kinds of reagents, the proc is still a hardcoded one so gotta figure out that one something (pickweight from list?)

DrChef replied:
Couldn't I make a generic milk type for different alien critters, if you only want different reagents and not a milk?

DrChef replied:
Actually, I'm going to make everything a list so it looks cleaner.

ma44 reviewed (changes requested):
Needs some proc rewrites + multiple reagent support for milking

ma44 commented on code/modules/food_and_drinks/food/snacks_other.dm:
```diff
@@ -559,7 +579,7 @@
 	filling_color = "#F6CB0B"
 	tastes = list("pineapple" = 1)
-	foodtype = FRUIT | PINEAPPLE
+	foodtype = FRUIT
```
🤔

DrChef replied:
It may of been from a earlier version of the source because I noticed there were some changes and I just copied my source code, where I should of just cut and pasted it in places instead of the whole file. I forked the latest version just moments ago so it probably had changes from the version I downloaded earlier today.

ma44 commented:
Actually this could be made into a component later on.

grhmhome committed:
Updated snacks_other.dm correctly.

I made a beginner mistake and fixed it by only cutting and pasting relevant code, not the whole file because possible conflicts.

grhmhome committed:
Update drink_reagents.dm

removed random text typo at top of page

DrChef commented:
I have some ideas what I could do with gutlunch.dm

I could change the milktype var to be cream and change line 135 to `reagents.add_reagent("[milktype]", rand(2, 5))`

Option 2 would make it where in line 321 in farm_animals.dm the milktype would be a list var, so I could make it where future critters that could be milked would have multiple reagents and then I would be able to put different reagents such as [milktype[0], [milktype[1], [milktype[2], but that would require me to change farm_animals.dm. I can do that, but I want to know what the coders think. What solution would be better? I like option 1 myself, but I want to know what you people think is a cleaner solution?

grhmhome committed:
updated gutlunch.dm with something that isnt a placeholder

Changed line 126 from "milk" to "cream" since milktype doesn't need a milk subtype. Any reagent will do for now. I decided to to change line 135 so instead of it using the reagent cream when cream is already a milktype already exists which is also cream, I changed "cream" to "[milktype]"

Fox-McCloud commented on code/modules/food_and_drinks/recipes/food_mixtures.dm:
```diff
@@ -87,6 +87,17 @@
+
+/datum/chemical_reaction/fetacheesewheel/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
+	for(var/i = 1, i <= created_volume, i++)
```
the `for(var/i in 1 to X)` format is preferred, as it's easier to read and faster.

ExcessiveUseOfCobblestone replied:
^

DrChef replied:
I'll make the required changes today.

Grazyn commented:
Greek salad, obviously. 

also
>feta
>tastes sweet

ExcessiveUseOfCobblestone commented on code/modules/mob/living/simple_animal/friendly/farm_animals.dm:
```diff
@@ -317,16 +317,21 @@
-	. = ..()
+        create_reagents(50)
+        reagents.add_reagent("[milktype]", 20)
+        . = ..()
```
this has 0 effect currently but put this as the first line under Initialize in case someone adds early returns.

SpinnerMaster commented:
Item 4 could be a gyro or something?

ExcessiveUseOfCobblestone commented:
leave it to ma44 to want to be able to milk various mobs.

epochayur commented:
"what do you do in your spare time?"
"i code virtual udders"

praisenarsie commented:
will i still be able to make butter

DaxDupont commented:
fuck me I need to buy feta tomorrow now

DrChef commented:
SpinnerMaster, I can make a gyro. Meat, flattened dough, feta, and tomato.
List of food items:
Greek salad, gyro, greek feta burger, feta ravioli, greek fries.

ExcessiveUseOfCobblestone commented:
Don't add gyros before you add togas!!!!!!!

SACRIFICE YOUR GYROS TO DIONYSUS

DrChef commented:
Greek Station 13

ShizCalev commented on code/modules/food_and_drinks/food/snacks_other.dm:
```diff
@@ -21,6 +21,25 @@
+
+/obj/item/reagent_containers/food/snacks/fetacheesslice
+	name = "feta cheese slice"
+	desc = "A wedge of delicious Feta. The cheese wheel it was cut from can't have gone far."
```
`couldn't have gone far`*

`Can't have` is an indicative mood, stating an absolute fact about a situation.
`Couldn't have` on the contrary is a subjunctive mood; stating an opinion or an attitude towards a hypothetical or (possibly) counterfactual situation or condition.

ShizCalev replied:
Also lowercase feta.

ShizCalev commented on code/modules/food_and_drinks/food/snacks_other.dm:
```diff
@@ -21,6 +21,25 @@
 
+/obj/item/reagent_containers/food/snacks/store/fetacheesewheel
+	name = "feta cheese wheel"
+	desc = "A big wheel of delcious Feta."
```
Lowercase feta

grhmhome committed:
adds line to snacks_other.dm

grhmhome committed:
Update snacks_other.dm

grhmhome committed:
Update snacks_other.dm

grhmhome committed:
grammar fix to cheese wedge and feta cheese slice

adds suggested grammar fixes

grhmhome committed:
changes text from sweet to tangy in snacks_other.dm

grhmhome committed:
corrects typo in snacks_other.dm coudln't to couldn't

DrChef commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm:
```diff
@@ -131,7 +132,7 @@
 /obj/item/udder/gutlunch/generateMilk()
 	if(prob(60))
-		reagents.add_reagent("cream", rand(2, 5))
+		reagents.add_reagent("[milktype]", rand(2, 5))
```
I was told that using a list instead for critters with multiple reagents that could be milked, so I was thinking of changing to that, so I'll change it.

DrChef replied:
I found a extremely simple way of making a list by doing something like snacks.dm

grhmhome committed:
Update snacks_other.dm

monkeyboyinc commented:
Oh god, the cheese floodgates have opened. A few months from now, we're going to turn into Dungeons of Dredmor.

Bawhoppen commented:
We can't have greekstation13 if Nanotrasen isn't in debt.

ExcessiveUseOfCobblestone commented:
Well the stock market perpetually crashed so we're right at home.

ExcessiveUseOfCobblestone commented:
Just a FYI it may be easier to toss this as a component rather than an organ.

Not sure how functional organs are with simple_mobs or if that's something we even want to bother with outside of butcher results.

ma44 commented:
The organs are processed with the simple_mobs lifetick, requires putting a proc in the life() in the first place for the organ to actually do stuff

optimumtact commented:
There's a bunch of unresponded to review comments, assuming this is stale
